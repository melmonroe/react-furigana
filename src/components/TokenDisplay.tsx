import Ruby from './Ruby.tsx';
import {Token} from '../tokenizer.ts';

type TokenDisplayProps = {
    token: Token;
}

export default function TokenDisplay(props: TokenDisplayProps) {

    if (props.token.furigana) {
        return <Ruby kanji={props.token.kanji} furigana={props.token.furigana}/>;
    }

    return props.token.kanji;

}
