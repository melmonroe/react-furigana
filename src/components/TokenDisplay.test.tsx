import {render} from '@testing-library/react';
import {describe, expect, it} from 'vitest';
import TokenDisplay from './TokenDisplay.tsx';

describe('TokenDisplay', () => {

    it('renders without furigana', () => {
        expect(render(
            <TokenDisplay token={{kanji: '漢字'}}/>
        )).toMatchSnapshot();
    });

    it('renders with furigana', () => {
        expect(render(
            <TokenDisplay token={{kanji: '漢字', furigana: 'かんじ'}}/>
        )).toMatchSnapshot();
    });

});
