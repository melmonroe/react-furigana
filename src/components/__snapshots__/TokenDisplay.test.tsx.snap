// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TokenDisplay > renders with furigana 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ruby>
        漢字
        <rt>
          かんじ
        </rt>
      </ruby>
    </div>
  </body>,
  "container": <div>
    <ruby>
      漢字
      <rt>
        かんじ
      </rt>
    </ruby>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TokenDisplay > renders without furigana 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      漢字
    </div>
  </body>,
  "container": <div>
    漢字
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
